pkg/hostagent/setup.go

```diff
@@ -306,21 +317,26 @@ func (agent *HostAgent) configureContainerIfaces(metadata *md.ContainerMetadata)
 		}
 		// if no mac is assigned, set it to the default Mac.
 		if len(iface.Mac) == 0 {
+			logger.Infof("No Mac assigned, assigning default")
 			iface.HostVethName, iface.Mac, err =
 				runSetupVeth(iface.Sandbox, iface.Name, agent.config.InterfaceMtu, nil)
 			if err != nil {
+				deallocIP(iface)
 				return nil, err
 			}
 		}
 
+		if len(iface.HostVethName) == 0 || len(iface.Mac) == 0 {
+			deallocIP(iface)
+			return nil, fmt.Errorf("Unable to Configure Container Interface")
+		}
+
 		agent.addToResult(iface, ifaceind, result)
 
 		logger.Debug("Configuring network for ", iface.Name, ": ", *result)
 		err = runSetupNetwork(iface.Sandbox, iface.Name, result)
 		if err != nil {
-			agent.ipamMutex.Lock()
-			agent.deallocateIpsLocked(iface)
-			agent.ipamMutex.Unlock()
+			deallocIP(iface)
```

Comment:
I think it will help to error log here that "runSetupNetwork" failed (perhaps also the err) and that we are returning.